test/test_sanitizer_log_parser.py

```diff
@@ -0,0 +1,78 @@
+from csv import DictReader
+import os
+from typing import Dict, Optional, Tuple
+
+from colcon_sanitizer_reports.sanitizer_log_parser import SanitizerLogParser
+import pytest
+
+
+class Fixture:
+
+    def __init__(self, resource_name: str) -> None:
+        self.resource_name = resource_name
+        self._parser: Optional[SanitizerLogParser] = None
+
+    @property
+    def resource_path(self) -> str:
+        return os.path.dirname(os.path.abspath(__file__)) + '/resources/' + self.resource_name
+
+    @property
+    def input_log_path(self) -> str:
+        return self.resource_path + '/input.log'
+
+    @property
+    def expected_output_csv_path(self) -> str:
+        return self.resource_path + '/expected_output.csv'
+
+    @property
+    def parser(self) -> SanitizerLogParser:
+        if self._parser is None:
+            parser = SanitizerLogParser()
+            parser.set_package(self.resource_name)
+            with open(self.input_log_path, 'r') as input_log_f_in:
+                for line in input_log_f_in.readlines():
+                    parser.add_line(line)
+
+            self._parser = parser
+            parser.set_package(None)
+
+        return self._parser
+
+    @property
+    def report_csv(self) -> DictReader:
+        return DictReader(self.parser.csv.split('\n'))
+
+    @property
+    def expected_csv(self) -> DictReader:
+        with open(self.expected_output_csv_path, 'r') as expected_output_csv_f_in:
+            return DictReader(expected_output_csv_f_in.read().split('\n'))
+
+
+@pytest.fixture(params=os.listdir(os.path.dirname(os.path.abspath(__file__)) + '/resources'))
```

Comment:
using `os.listdir` is very dangerous, why not list test files explicitly? this will avoid having this test runner pick up whatever happens to be placed in the dir (think about temporary files from your editor, etc.).

```py
_TEST_FILES=[
   'a/b/cd.csv',
   'a/b/cd.csv'
  # etc.
]